backend/cmm_helpers.mli

```diff
@@ -1226,3 +1226,149 @@ val set_field_unboxed :
 val dls_get : dbg:Debuginfo.t -> expression
 
 val poll : dbg:Debuginfo.t -> expression
+
+module Scalar_type : sig
+  (** A static_cast from a larger integral type to a smaller one logically truncates the
+      upper bits. Note that values are stored in registers sign- or zero- extended
+      according to their signdness, so the result may be sign-extended.
+
+      A static_cast from a smaller integral type to an equal or larger sized-integral type
+      sign- or zero-extends the input value according to the sign of the result.
+
+      A static_cast from an integral type to a float is pretty self-explanatory.
+
+      A static_cast from a float to an integral type always rounds toward zero. If the
+      resulting integral does not fit in the destination type, the result is unspecified
+      (although it's generally zero).
+
+      Casting floats to/from unsigned register-width integers is not implemented and will
+      raise in the compiler. *)
+
+  type 'a static_cast :=
+    dbg:Debuginfo.t -> src:'a -> dst:'a -> expression -> expression
+
+  (** Conjugate f by [static_cast ~src:outer ~dst:inner].
+
+      Shorthand for:
+      - [static_cast] the argument from [outer] to [inner]
+      - apply [f]
+      - [static_cast] back from [inner] to [outer] *)
+  type 'a conjugate :=
+    outer:'a ->
+    inner:'a ->
+    dbg:Debuginfo.t ->
+    f:(expression -> expression) ->
+    expression ->
+    expression
+
+  (** An IEEE 754 floating-point number *)
+  module Float_width : sig
+    type t = Cmm.float_width =
+      | Float64
+      | Float32
+
+    val static_cast : t static_cast
+  end
+
+  module Signedness : sig
+    type t =
+      | Signed
+      | Unsigned
+
+    val equal : t -> t -> bool
+
+    val print : Format.formatter -> t -> unit
+  end
+
+  module type Integral_ops := sig
+    type t
+
+    val print : Format.formatter -> t -> unit
+
+    val equal : t -> t -> bool
+
+    val signedness : t -> Signedness.t
+
+    val with_signedness : t -> signedness:Signedness.t -> t
+
+    val signed : t -> t
+
+    val unsigned : t -> t
+
+    val can_cast_without_losing_information : src:t -> dst:t -> bool
```

Comment:
Some more documentation in general for these modules would be good, but at the very least for this function, you should document that even if it returns `[true]` it does not necessarily mean that the cast (that does not lose information) is the identity (since there may be the need to tag/untag).